x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/function/scalar/ScalarFunction.java

```diff
@@ -108,19 +109,28 @@ protected ScriptTemplate scriptWithScalar(ScalarFunction scalar) {
     }
     
     protected ScriptTemplate scriptWithAggregate(AggregateFunction aggregate) {
-        String template = "{}";
+        String template = basicTemplate(aggregate);
         return new ScriptTemplate(processScript(template),
                 paramsBuilder().agg(aggregate).build(),
                 dataType());
     }
 
     protected ScriptTemplate scriptWithGrouping(GroupingFunction grouping) {
```

Comment:
Shouldn't we remove this method as currently there is no such case that a groupingfunction (actually histogram) results into a script?

Reply:
Chose to keep it there since, before the bug was introduced, the code was using `asDateTime` for grouping functions, as well.

Reply:
Would you mind adding a comment then, please, to avoid pursuing it again in the future?